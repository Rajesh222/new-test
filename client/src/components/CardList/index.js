import React, { Component } from 'react';

class CardList extends Component {
  renderCardlist = () => {
    const { allCardList } = this.props;
    return allCardList.map(item => {
      return (
        <div className='card-list' key={item.id}>
          <div className='item'>{item.name}</div>
          <div className='item'>{item.cardNumber}</div>
          <div className='item'>£{item.balance}</div>
          <div className='item'>£{item.limit}</div>
        </div>
      );
    });
  };
  render() {
    const { allCardList } = this.props;
    return (
      <div className='list'>
        <h5>Existing Cards</h5>
        <div className='card-list header'>
          <div className='item'>Name</div>
          <div className='item'>Card Number</div>
          <div className='item'>Balance</div>
          <div className='item'>Limit</div>
        </div>
        {allCardList.length > 0 && this.renderCardlist()}
      </div>
    );
  }
}

export default CardList;
