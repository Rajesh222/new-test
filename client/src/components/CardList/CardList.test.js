import React from 'react';
import { mount, shallow } from 'enzyme';
import CardList from './index';

describe('CardList component', () => {
    it('renders correctly', () => {
        const allCardList = [];
        const wrapper = shallow(<CardList allCardList={allCardList} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('has list classname', () => {
        const allCardList = [];
        const wrapper = shallow(<CardList allCardList={allCardList} />);
        expect(wrapper.find('.list').length).toBe(1);
        expect(wrapper).toMatchSnapshot();
    });

    it('should have eqaual number of credit card number record', () => {
        const allCardList = [{
            id: 1,
            name: 'test',
            cardNumber: '123',
            limit: 1000,
            balance:100
        }];
        const wrapper = shallow(<CardList allCardList={allCardList} />);
        expect(wrapper.find('.card-list').length).toBe(2);
        expect(wrapper).toMatchSnapshot();
    });

});
