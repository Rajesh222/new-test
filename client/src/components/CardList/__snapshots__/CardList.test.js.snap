// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component has list classname 1`] = `ShallowWrapper {}`;

exports[`CardList component renders correctly 1`] = `ShallowWrapper {}`;

exports[`CardList component should have eqaual number of credit card number record 1`] = `ShallowWrapper {}`;
